sensu/plugins/metrics-keystone.py

```diff
@@ -0,0 +1,63 @@
+#/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+
+import time
+from collections import namedtuple
+
+import shade
+
+OK = 0
+WARNING = 1
+CRITICAL = 2
+UNKNOWN = 3
+
+
+class CloudMetrics(object):
+    def __init__(self):
+        self.cloud = shade.openstack_cloud()
+
+    def num_projects(self):
+        return len(self.cloud.keystone_client.tenants.list())
```

Comment:
It's essentially the same thing.  `search_projects` calls `list_projects`, this just provides filtering.